tests/test_init_and_backend.py

```diff
@@ -0,0 +1,69 @@
+import os
+from os import error
+import unittest
+from pathlib import Path
+import sys
+import tempfile
+import shutil
+import filecmp
+
+sys.path.insert(0, os.getcwd() +'/../')
+import mkosi.backend as backend
+import mkosi.__init__ as init
+
+class TestMkosiMethods(unittest.TestCase):
+
+    #test relative paths with /../../
+    def test_workspace(self):
+        self.assertEqual(Path('/home/folder/mkosi'), backend.workspace(Path('/home/folder/mkosi/mkosi')))
+        self.assertEqual(Path('/home/../home/folder/mkosi'), backend.workspace(Path('/home/../home/folder/mkosi/mkosi')))
+        self.assertEqual(Path('/'), backend.workspace(Path('/')))
+        self.assertEqual(Path(), backend.workspace(Path()))
+
+    #actually use fedora keys
+    def test_fedora_release_cmp(self):
+        self.assertEqual(0, init.fedora_release_cmp("rawhide", "rawhide"))
+        self.assertEqual(0, init.fedora_release_cmp("32", "32"))
+        self.assertTrue(init.fedora_release_cmp("33", "32") > 0)
+        self.assertTrue(init.fedora_release_cmp("30", "31") < 0)
+        self.assertTrue(init.fedora_release_cmp("-1", "-2") > 0)
+        self.assertTrue(init.fedora_release_cmp("1", "-2") > 0)
+        with self.assertRaises(ValueError):
+            init.fedora_release_cmp("literal", "rawhide")
+
+
+    def test_strip_suffixes(self):
+        self.assertEqual(Path("home/test"), init.strip_suffixes(Path("home/test.zstd")))
+        self.assertEqual(Path("home/test"), init.strip_suffixes(Path("home/test.xz")))
+        self.assertEqual(Path("home/test"), init.strip_suffixes(Path("home/test.raw")))
+        self.assertEqual(Path("home/test"), init.strip_suffixes(Path("home/test.tar")))
+        self.assertEqual(Path("home/test"), init.strip_suffixes(Path("home/test.cpio")))
+        self.assertEqual(Path("home/test"), init.strip_suffixes(Path("home/test.qcow2")))
+        self.assertEqual(Path("home.xz/test"), init.strip_suffixes(Path("home.xz/test.xz")))
+        self.assertEqual(Path("home.xz/test"), init.strip_suffixes(Path("home.xz/test")))
+
+
+    def test_copy_file(self):
+        temp1 = tempfile.NamedTemporaryFile(delete=True)
+        temp2 = tempfile.NamedTemporaryFile(delete=True)
```

Comment:
I think this should better be used in a `with` block. `delete=True` can be left out, since it is the default.